dmscripts/notify_suppliers_of_awarded_briefs.py

```diff
@@ -0,0 +1,99 @@
+from datetime import datetime, timedelta
+
+from dmutils.email.exceptions import EmailError
+from dmutils.email.helpers import hash_string
+from dmutils.formats import DATE_FORMAT
+from dmscripts.helpers import env_helpers
+
+
+def _create_context_for_brief(stage, brief):
+    return {
+        'brief_title': brief['title'],
+        'brief_link': '{}/{}/opportunities/{}'.format(
+            env_helpers.get_web_url_from_stage(stage),
+            brief['framework']['family'],
+            brief['id']
+        )
+    }
+
+
+def _get_brief_responses_to_be_notified(data_api_client, brief_response_ids, awarded_at):
+    brief_responses = []
+
+    if brief_response_ids:
+        # Used for re-running failed email sending
+        for brief_response_id in brief_response_ids:
+            brief_responses.append(data_api_client.get_brief_response(brief_response_id)['briefResponses'])
+    else:
+        awarded_at_date = awarded_at or (datetime.utcnow() - timedelta(days=1)).strftime(DATE_FORMAT)
+        awarded_brief_responses = data_api_client.find_brief_responses_iter(awarded_at=awarded_at_date)
+
+        for abr in awarded_brief_responses:
+            # Add the successful BriefResponse
+            brief_responses.append(abr)
+            # Get the unsuccessful BriefResponses
+            submitted_brief_responses = data_api_client.find_brief_responses_iter(
+                brief_id=abr['briefId'], status='submitted'
+            )
+            brief_responses.extend(submitted_brief_responses)
+
+    return brief_responses
+
+
+def _build_and_send_emails(brief_responses, mail_client, stage, dry_run, template_id, logger):
+    failed_brief_responses = []
+
+    # Now email everyone whose Brief got awarded
+    for brief_response in brief_responses:
+        email_address = brief_response["respondToEmailAddress"]
+        if not email_address:
+            continue
+
+        brief_email_context = _create_context_for_brief(stage, brief_response['brief'])
+        try:
+            if not dry_run:
+                mail_client.send_email(
+                    email_address, template_id, brief_email_context, allow_resend=False
```

Comment:
What error does this function give us if it _does_ turn out to be a resend? Are we able to discern that error from a real error? Should we log it differently?

Reply:
Just to clarify, there aren't automatic resends. It will be a manual triggering of Jenkins with either the ID list or a date string. So we should be aware that something unusual is happening.